traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ParkingSearchResultAdapter.java

```diff
@@ -0,0 +1,258 @@
+                itemHolder.selectionIndicator.setBackgroundColor(color);
+
+                itemHolder.viewButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        parkingSearchItemClickListener.onParkingItemDetailsClick(parkingItem);
+                    }
+                });
+                break;
+            case HEADER_VIEW_TYPE:
+                HeaderViewHolder headerHolder = (HeaderViewHolder) holder;
+                headerHolder.textView.setText(String.format(Locale.getDefault(), "%d Parking Lots Found", parkingItemSearchResult.getItems().size()));
+                break;
+            case FOOTER_VIEW_TYPE:
+            case LOADING_VIEW_TYPE:
+                break;
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        List itemsList = parkingItemSearchResult.getItems();
+        if (orientation == LinearLayoutManager.VERTICAL) {
+            if (position == 0) {
+                return HEADER_VIEW_TYPE;
+            } else {
+                int reindexedPosition = position - 1;
+                if (reindexedPosition >= itemsList.size() || itemsList.get(reindexedPosition) == null) {
+                    this.onFetchItems(reindexedPosition);
+                    return LOADING_VIEW_TYPE;
+                } else {
+                    return PARKING_ITEM_VIEW_TYPE;
+                }
+            }
+
+        } else if (orientation == LinearLayoutManager.HORIZONTAL) {
+            if (position >= itemsList.size()) {
+                return FOOTER_VIEW_TYPE;
+            } else {
+                if (itemsList.get(position) == null) {
+                    this.onFetchItems(position);
+                    return LOADING_VIEW_TYPE;
+                } else {
+                    return PARKING_ITEM_VIEW_TYPE;
+                }
+            }
+        }
+        throw new IllegalArgumentException("unknown item type at position:" + position +
+                " in orientation:" + orientation);
+    }
+
+    @Override
+    public int getItemCount() {
+        //adds one for header (vertical "23 parking lots found") or footer (horizontal "search more")
+        return 1 + (parkingItemSearchResult != null ? parkingItemSearchResult.getTotal() : 0);
+    }
+
+    public void setSelectedParkingItem(ParkingItem parkingItem) {
+        if (selectedParkingItem != null) {
+            int previousIndex = parkingItemSearchResult.getItems().indexOf(selectedParkingItem);
+            selectedParkingItem = null;
+            notifyItemChanged(previousIndex);
+        }
+        selectedParkingItem = parkingItem;
+        int newIndex = parkingItemSearchResult.getItems().indexOf(selectedParkingItem);
+        notifyItemChanged(newIndex);
+        parkingSearchItemClickListener.scrollListToIndex(newIndex);
```

Comment:
language here is bad,..it shouldnt speak "SCROLL"...it just tells the delegate it has been selected...if it scrolls or not is its business

Reply:
moved the responsibility for scrolling list to parkingActivity, makes more sense for parkingActivity rather than the adapter to control it (the adapter doesnt hold the linear layout manager)

Reply:
✓